Source/JavaScriptCore/wasm/WasmOMGIRGenerator.cpp

```diff
@@ -5880,8 +5880,14 @@ auto OMGIRGenerator::emitDirectCall(unsigned callProfileIndex, FunctionSpaceInde
                 handle->collectStackMap(this, params);
 
             ASSERT(!m_info.isImportedFunctionFromFunctionIndexSpace(functionIndexSpace));
-            Ref<IPIntCallee> callee = m_calleeGroup.ipintCalleeFromFunctionIndexSpace(functionIndexSpace);
-            jit.storeWasmCalleeToCalleeCallFrame(CCallHelpers::TrustedImmPtr(CalleeBits::boxNativeCallee(callee.ptr())), isTailCallRootCaller ? sizeof(CallerFrameAndPC) - prologueStackPointerDelta() : 0);
+
+            // If the call is self-recursion, it is guaranteed that callee will be OMG as well since it ends up calling itself.
+            // Since the OMG function prologue will put callee, the caller does not need to place it.
+            bool selfRecursion = &m_inlineRoot->m_info == &m_info && m_info.toCodeIndex(functionIndexSpace) == m_inlineRoot->m_functionIndex;
```

Comment:
In general, if the callee in anything other than ipint, we can skip this right?

Reply:
Hmm, I guess our repatching code sets callee first? What about if our callee code is in the jit region?

Reply:
Caller is always setting an IPIntCallee, and we are no longer repatching callee for calls.